### DISCUSSION TITLE
Azure Media CDN not clear

### DISCUSSION THREAD

**Discussion by webmedia1012** [OrchardCMS/OrchardCore#7076] [Q&A] [answered]
Hi good people!
I have a quick question, the documentation for handling media on Azure CDN is very simple but then quite incomplete. Here's what I mean, after following the instruction of adding configuration on appsettings.json,  the images still doesn't load, not sure if the appsettings.json config is the one on root of the project or the app_data folder, if it is, then it should clearly define it, it is only mentioned as this "The feature replaces the default App_Data file based media store with an Azure Media Storage Provider." 

Anyway, what would be the best approach?, and once properly configured, would this mean that if you upload via media on content items, it would be saved on CDN ? and not "app_data" folder? or you have to manually transfer the media to the CDN before i can be accessed.

As always thank you for the prompt response.

Regards!
G

**Comment by deanmarcussen** [comment, marked as answer]
Ok, so lets clear a few things up.

Azure CDN is a different thing to Azure Blob Storage.

- Azure Blob Storage, and the feature we offer for it will replace the default `App_Data` file system for Media only.

- You will have to transfer your existing media from `App_Data` to Azure Blob Storage.
- New media will be uploaded to Blob Storage

It is not a CDN, just an external storage provider.

Media will still be served by the host (Orchard Core) site, but Orchard Core will fetch and cache images from Blob Storage as required.

If you want to use a CDN we have an option for configuring a url prefix, and you would point your CDN at the Orchard Core site (not blob storage)

Configuration can be done in either the `tenants/appsettings.json` on a per tenant basis, or in your main `appsettings.json` file.

See docs here on how these options can be configured. https://docs.orchardcore.net/en/dev/docs/reference/core/Configuration/

The easiest way to test if you have correctly configured a connection string is to go to the `Media Cache` feature after enabling blob storage and you will see an error message if the connection string is not correctly configured.

If you're still having difficulty configuring, please upload the appsettings.json configuration, with details on which one it is (i.e. main host, or tenants)

**Reply by webmedia1012** [reply]
Hi @deanmarcussen, it makes sense, thank you so much as always for the elaborate explanation,   so to be clear,before the blob storage configuration the way we point to media is "domain.com/media/assetfile.png", you're saying after enabling the azure storage blob, it will be https://storage.blob.core.windows.net/containerName/Media/assetfile.png and not the CDN https://name.azureedge.net/media/assetfile.png. This would mean changing all contents pointing to previous asset "domain.com/media/assetfile.png). And yes I do understand I have to move the files from app_data\media folder to the storage blob.

Thank you again.

**Reply by deanmarcussen** [reply]
Nope, that's not what I'm saying at all :)

Before blob storage the url will be `domain.com/media/assetfile.png?width=200`
After blob the url will be `domain.com/media/assetfile.png?width=200`

If you use a CDN (which is totally unrelated to blob storage) you can use the `CdnBaseUrl` media setting.

When you use this setting you point your CDN at `domain.com` not the blob url.

All of this is essential to being able to serve resized images.

The CDN setting works well with blob storage or normal storage. 

They are unrelated.

Do not connect your CDN to the blob storage account connect it to the url of the site

**Reply by webmedia1012** [reply]
I think I got it ! thank you very much for clarifying that.
